src/wmtk/operations/EdgeCollapse.hpp

```diff
@@ -26,28 +26,20 @@ class EdgeCollapse : public MeshOperation
         const wmtk::operations::CollapseBasicStrategy& strategy =
             wmtk::operations::CollapseBasicStrategy::Default);
 
-protected:
-    std::vector<simplex::Simplex> execute_aux(EdgeMesh& mesh, const simplex::Simplex& simplex)
-        override;
-    std::vector<simplex::Simplex> unmodified_primitives_aux(
-        const EdgeMesh& mesh,
-        const simplex::Simplex& simplex) const override;
-
-    std::vector<simplex::Simplex> execute_aux(TriMesh& mesh, const simplex::Simplex& simplex)
-        override;
-    std::vector<simplex::Simplex> unmodified_primitives_aux(
-        const TriMesh& mesh,
-        const simplex::Simplex& simplex) const override;
-
-    std::vector<simplex::Simplex> execute_aux(TetMesh& mesh, const simplex::Simplex& simplex)
-        override;
-    std::vector<simplex::Simplex> unmodified_primitives_aux(
-        const TetMesh& mesh,
+    std::vector<simplex::Simplex> execute(const simplex::Simplex& simplex) override;
+    std::vector<simplex::Simplex> unmodified_primitives(
         const simplex::Simplex& simplex) const override;
 
 private:
     std::vector<std::shared_ptr<operations::BaseCollapseNewAttributeStrategy>>
         m_new_attr_strategies;
+
+    template <typename MeshType>
+    std::vector<simplex::Simplex> execute_tmpl(MeshType& mesh, const simplex::Simplex& simplex);
```

Comment:
why these?

Reply:
i'll delete, was an earlier attempt to simplify teh code